### PR TITLE
PJ adjustable pool sizes

### PR DESCRIPTION
**Pull request by andreas-wehrmann**
This can be used to deal with memory fragmentation issues in long running applications that encounter temporary high loads.

For reference, see this issue:
https://github.com/pjsip/pjproject/issues/2389

### CODE REVIEW

File: pjmedia/include/pjmedia/config.h [hunk 1]
```diff
@@ -44,6 +44,34 @@
...
+#   define PJMEDIA_POOL_EVTMGR_LEN		500
+#endif
+
+/**
+ * Memory increment for evnt manager.
+ */
+#ifndef PJMEDIA_POOL_EVTMGR_INC
+#   define PJMEDIA_POOL_EVTMGR_INC		500
+#endif
+
```
Comment (nanangizz) [inline pjmedia/include/pjmedia/config.h:74]: Sorry just got a chance to revisit this PR.

IIRC we tried to reproduce the memory fragmentation issue (can't remember the result, can't find any doc/mail history either :D ). Anyway, some pools have been configurable already, so I think it is okay to integrate the patch, just perhaps better to try to maintain a consistent config name pattern, which mostly seem to use: `[MODULE]_POOL_LEN/INC_[OBJECT]`, e.g: `PJSIP_POOL_LEN/INC_TRANSPORT`.

Comment (andreas-wehrmann) [reply pjmedia/include/pjmedia/config.h:74]: Hello there,

I tried to adhere to the name patterns I found the in modules in which I made the changes; but I'll go and revisit the changes to see if I missed something.

As for the fragmentation issue:
I specifically made only the pools configurable which I found to be problematic in my case.
I experimented with cleaning up pools under certain conditions (which definitely helped with the fragmentation issue)
but I wasn't happy with what I came up and stayed with the mitigation enabled by the changes proposed in this PR.

File: pjmedia/src/pjmedia-codec/g722.c [hunk 2]
```diff
@@ -174,8 +183,8 @@ PJ_DEF(pj_status_t) pjmedia_codec_g722_init( pjmedia_endpt *endpt )
     g722_codec_factory.endpt = endpt;
     g722_codec_factory.pcm_shift = PJMEDIA_G722_DEFAULT_PCM_SHIFT;
 
-    g722_codec_factory.pool = pjmedia_endpt_create_pool(endpt, "g722", 1000, 
-						        1000);
+    g722_codec_factory.pool = pjmedia_endpt_create_pool(endpt, "g722", G722_POOL_INIT_SIZE, 
+						        G722_POOL_INC_SIZE);
```
Comment (nanangizz) [inline pjmedia/src/pjmedia-codec/g722.c:187]: Sorry, forgot to mention this earlier.
- Perhaps better to put this setting in `pjmedia/include/pjmedia-codec/config.h`? Named `PJMEDIA_CODEC_POOL_LEN/INC_G722`?
- Also for all changes in the library code, please use max column length of 80.

Thank you.

Comment (andreas-wehrmann) [reply pjmedia/src/pjmedia-codec/g722.c]: Putting a general define like `PJMEDIA_CODEC_POOL_LEN` in `pjmedia/include/pjmedia-codec/config.h` would imply, that every codec uses it, no?
If I introduce this define, should I update every codec impl to make use of it?
If every codec uses it, what would a good default value be?

Comment (nanangizz) [reply pjmedia/src/pjmedia-codec/g722.c]: > Putting a general define like `PJMEDIA_CODEC_POOL_LEN` in `pjmedia/include/pjmedia-codec/config.h` would imply, that every codec uses it, no?

Was thinking for G722 only, as shown in the patch. But, having a common setting for all codecs sounds like a good idea.

> If I introduce this define, should I update every codec impl to make use of it?

Yes, unfortunately.

> If every codec uses it, what would a good default value be?

Assuming each codec instance will need to allocate about 100-500bytes (mostly for PJMEDIA codec structures, as some codec backends allocate their states by theirselves), IMO default value in the range of 1-5kB should be fine, and 2kB is perhaps reasonable as max calls in PJSUA by default is 4 (so 4*500B = 2kB).

Feel free if you have any opinions or suggestions.

Comment (andreas-wehrmann) [reply pjmedia/src/pjmedia-codec/g722.c]: Moved and renamed the defines to pjmedia-codec's config.h now;
Having a define for every codec implementation allows for more fine tuning, if that should become necessary.

Comment (andreas-wehrmann) [reply pjmedia/src/pjmedia-codec/g722.c]: Did you get a chance to look at the new changes?
I'm not sure whether I need to explicitly request a review or you get notified automatically.

Comment (nanangizz) [reply pjmedia/src/pjmedia-codec/g722.c]: Yes, we get notified automatically. Sorry for the delay, must be in the middle of something when the notification came, but then forgot to come back to check it.

Things look okay to me, need to wait for another reviewer (from PJSIP team) to be able to merge.

**Pull request merged**